tools/ci_build/github/azure-pipelines/linux-gpu-ci-pipeline.yml

```diff
@@ -13,7 +13,7 @@ jobs:
     parameters:
       Dockerfile: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_cuda11
       Context: tools/ci_build/github/linux/docker
-      DockerBuildArgs: "--network=host --build-arg POLICY=manylinux2014 --build-arg PLATFORM=x86_64 --build-arg BASEIMAGE=nvcr.io/nvidia/cuda:11.1.1-cudnn8-devel-centos7 --build-arg DEVTOOLSET_ROOTPATH=/opt/rh/devtoolset-9/root --build-arg PREPEND_PATH=/opt/rh/devtoolset-9/root/usr/bin: --build-arg LD_LIBRARY_PATH_ARG=/opt/rh/devtoolset-9/root/usr/lib64:/opt/rh/devtoolset-9/root/usr/lib:/opt/rh/devtoolset-9/root/usr/lib64/dyninst:/opt/rh/devtoolset-9/root/usr/lib/dyninst:/usr/local/lib64 --build-arg BUILD_UID=$( id -u )"
+      DockerBuildArgs: "--network=host --build-arg POLICY=manylinux2014 --build-arg PLATFORM=x86_64 --build-arg BASEIMAGE=nvcr.io/nvidia/cuda:11.4.1-cudnn8-devel-centos8 --build-arg DEVTOOLSET_ROOTPATH=/opt/rh/devtoolset-9/root --build-arg PREPEND_PATH=/opt/rh/devtoolset-9/root/usr/bin: --build-arg LD_LIBRARY_PATH_ARG=/opt/rh/devtoolset-9/root/usr/lib64:/opt/rh/devtoolset-9/root/usr/lib:/opt/rh/devtoolset-9/root/usr/lib64/dyninst:/opt/rh/devtoolset-9/root/usr/lib/dyninst:/usr/local/lib64 --build-arg BUILD_UID=$( id -u )"
```

Comment:
might be a typo. should be centos7

Reply:
I was looking at [these options](https://ngc.nvidia.com/catalog/containers/nvidia:cuda/tags)

11.4.1-runtime-centos7 (11.4.1/centos7/runtime/Dockerfile)
11.4.1-devel-centos7 (11.4.1/centos7/devel/Dockerfile)
11.4.1-base-centos7 (11.4.1/centos7/base/Dockerfile)

11.4.1-cudnn8-runtime-centos8 (11.4.1/centos8/runtime/cudnn8/Dockerfile)
11.4.1-runtime-centos8 (11.4.1/centos8/runtime/Dockerfile)
11.4.1-cudnn8-devel-centos8 (11.4.1/centos8/devel/cudnn8/Dockerfile)
11.4.1-devel-centos8 (11.4.1/centos8/devel/Dockerfile)
11.4.1-base-centos8 (11.4.1/centos8/base/Dockerfile)

Reply:
Then they stopped supporting it.  I can add centos 8 support to our manylinux scripts though.

Reply:
no let's not switch yet.
we found something similar when updating the TensorRT manylinux dockerfile. even though there is no cudnn8 in the name, the build succeeds https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_cuda11_4_tensorrt8_0#L1
I think cudnn is in there.
actually, why don't we use the same one referenced in the trt8 manylinux dockerfile?

or update everything to 11.4.1-devel-centos7 as another option

Reply:
nvcr.io/nvidia/cuda:11.4.0-devel-centos7 doesn't have cudnn but 
nvidia/cuda:11.4.0-cudnn8-devel-centos7 does.

Reply:
can you use nvidia/cuda:11.4.0-devel-centos7   ?  
use the exact same as in https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_cuda11_4_tensorrt8_0#L1 I mentioned above.
That is driving the Linux-TensorRT-CI pipeline, which seems to be working fine.

Reply:
can you compare that dockerfile and the one you are trying to update? I'm confused why there would be a failure. 
they should be almost the same (other than installing TensorRT dependencies)

Reply:
Yes, 
- pulling nvcr.io/nvidia/cuda:11.4.0-devel-centos7 --> cudnn.h not found
- nvidia/cuda:11.4.0-devel-centos7 --> cudnn.h not found
-  nvidia/cuda:11.4.0-cudnn8-devel-centos7 --> ./usr/include/cudnn.h

Reply:
They shouldn't have this difference. Did I miss something?

Reply:
So the idea is to get base image from nvidia/cuda:11.4.0-devel-centos7  then install cudnn by following the instructions at https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html#package-manager-rhel-install ?

Reply:
yes, looks like we'll have to install cudnn ourselves.
Dockerfile.manylinux2014_cuda11_4_tensorrt8_0  works because when it installs tensorrt packages, it will install the cudnn8 dependency.


Installing:
 libnvinfer-devel            x86_64     8.0.1-1.cuda11.3         cuda     157 M
 libnvinfer-plugin-devel     x86_64     8.0.1-1.cuda11.3         cuda      12 M
 libnvinfer-plugin8          x86_64     8.0.1-1.cuda11.3         cuda      12 M
 libnvinfer8                 x86_64     8.0.1-1.cuda11.3         cuda     139 M
 libnvonnxparsers-devel      x86_64     8.0.1-1.cuda11.3         cuda     415 k
 libnvonnxparsers8           x86_64     8.0.1-1.cuda11.3         cuda     713 k
 libnvparsers-devel          x86_64     8.0.1-1.cuda11.3         cuda     617 k
 libnvparsers8               x86_64     8.0.1-1.cuda11.3         cuda     850 k
Installing for dependencies:
 cuda-nvrtc-11-3             x86_64     11.3.109-1               cuda      34 M
 cuda-nvrtc-devel-11-3       x86_64     11.3.109-1               cuda      28 k
 libcudnn8                   x86_64     8.2.2.26-1.cuda11.4      cuda     769 M
 libcudnn8-devel             x86_64     8.2.2.26-1.cuda11.4      cuda     583 M